### PR TITLE
feat(fibre)!: add min/max bounds for payment promise timeout and shard retention

### LINKED ISSUE #7504: set min/max values for shard retention and expiration
Just a quick question: should we have some min shard retention that we can't go lower, that still makes sense? I see that in other places we don't have that, so probably current solution is consistent, but just wanted to check this. Because if we put some value like 1 (also we put ExpiresAt = 1), then the protocol will not be working at all.

_Originally posted by @mcrakhman in https://github.com/celestiaorg/celestia-app/pull/7489#discussion_r3518266325_

AC:

- [ ] have reasonable min and max values for these params

### PR DESCRIPTION
**Pull request by rach-id**
Closes #7504
PROTOCO-2060

## Problem

`payment_promise_timeout` and `shard_retention` validated only that they were positive. As raised in [this review comment](https://github.com/celestiaorg/celestia-app/pull/7489#discussion_r3518266325), governance could set them to values like `1ms`, at which point the protocol would not work at all.

## Change

Both params are now range-checked:

| Param | Min | Max | Default |
|---|---|---|---|
| `payment_promise_timeout` | `10m` | `12h` | `1h` (unchanged) |
| `shard_retention` | `10m` | `168h` (7d) | `4h` (unchanged) |

`payment_promise_timeout` is bounded below so a promise stays valid long enough for the client to upload its shards, collect validator signatures, and get `MsgPayForFibre` into a block. It is bounded above so it stays well inside both the `withdrawal_delay` window gating how old its `creation_timestamp` may be and the `payment_promise_retention_window` record that prevents double settlement.

`shard_retention` is bounded below so shards outlive the window in which a client fetches them back, and above because retention is a local storage obligation on every assigned validator.

Both defaults fall inside the new ranges, so no migration is needed.

## Testing

Added `x/fibre/types/params_test.go` covering wrong type, nil, zero, negative, and each bound at ±1ns and exactly on it, plus a check that `DefaultParams()` validates. The boundary cases are meaningful guards: `10m-1ns` and `168h+1ns` are both strictly positive, so the previous `<= 0` check accepted them.

`make build`, `make test-short` (69/69 packages), and `golangci-lint run ./x/fibre/...` all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: x/fibre/types/params.go [hunk 1]
```diff
@@ -35,6 +35,22 @@ var (
...
 
+const (
+	// MinPaymentPromiseTimeout is the lower bound of the payment promise timeout
+	// parameter. A promise has to stay valid long enough for the client to upload
+	// its shards to the assigned validators, collect their signatures, and get
+	// MsgPayForFibre included in a block.
+	MinPaymentPromiseTimeout = 10 * time.Minute
+	// MaxPaymentPromiseTimeout is the upper bound of the payment promise timeout
+	// parameter.
+	MaxPaymentPromiseTimeout = 12 * time.Hour
```
[the remark this answers is not in the document]
Comment (rach-id) [reply x/fibre/types/params.go:46]: will be handled in a separate PR

File: x/fibre/types/params.go [hunk 2]
```diff
@@ -141,8 +157,12 @@ func validatePaymentPromiseTimeout(v any) error {
...
 	}
 
-	if *duration <= 0 {
-		return fmt.Errorf("payment promise timeout must be positive: %s", *duration)
+	if *duration < MinPaymentPromiseTimeout {
+		return fmt.Errorf("payment promise timeout must be at least %s: %s", MinPaymentPromiseTimeout, *duration)
+	}
+
+	if *duration > MaxPaymentPromiseTimeout {
+		return fmt.Errorf("payment promise timeout must be at most %s: %s", MaxPaymentPromiseTimeout, *duration)
```
Comment (ninabarbakadze) [inline x/fibre/types/params.go:165]: could this be a validation function since it's repeated twice and asserts the same rule? so we can avoid fragmentation

**Review by ninabarbakadze — APPROVED** [review]
lgtm one nit

File: x/fibre/types/params.go (hunk 2 above)
Comment (rach-id) [reply x/fibre/types/params.go:165]: where is it repeated? one is for PP timeout and one is for shard retention IIUC
